apps/studio/src/components/tableview/TableTable.vue

```diff
@@ -1726,7 +1726,41 @@ export default Vue.extend({
       this.detailViewTitle = `Row ${position}`
       this.selectedRow = row
       this.selectedRowIndex = position
-      this.selectedRowData = this.$bks.cleanData(data, this.tableColumns)
+      
+      // Clean the data first
+      let cleanedData = this.$bks.cleanData(data, this.tableColumns)
+      
+      // Parse JSON columns
+      this.tableColumns.forEach(column => {
+        const columnValue = cleanedData[column.field]
+
+        // Check if the column is a JSON column
+        let isJsonColumn = String(column.dataType).toUpperCase() === 'JSON' || String(column.dataType).toUpperCase() === 'JSONB'
+
+        // If the column is not a JSON column, check if it is a JSON string
+        if (!isJsonColumn) {
```

Comment:
If we end up checking the string if it's json or not, then why don't we just call `JSON.parse` all the time? That way it'd be a lot simpler :)

Reply:
Yeah, I can see that. Something like `"123"` on a text column would be converted to an actual number which is not good.